n3fit/src/n3fit/backends/keras_backend/operations.py

```diff
@@ -107,36 +106,29 @@ def numpy_to_tensor(ival, **kwargs):
 
 # f(x: tensor) -> y: tensor
 def batchit(x, batch_dimension=0, **kwarg):
-    """ Add a batch dimension to tensor x """
+    """Add a batch dimension to tensor x"""
     return tf.expand_dims(x, batch_dimension, **kwarg)
 
 
 # layer generation
-def numpy_to_input(numpy_array, no_reshape=False, name=None):
+def numpy_to_input(numpy_array: np.ndarray, name: str = None):
```

Comment:
```suggestion
def numpy_to_input(numpy_array: npt.NDArray, name: str | None = None):
```
If type checking is to be adopted, why not use the [numpy.typing](https://numpy.org/devdocs/reference/typing.html)? A pickier point concerns the assignment of  `None` to `str`.

Reply:
Thanks I didn't know about `|`. Actually I see this only works from python 3.10, and another way to express this is `Optional[str]`, so I'll replace everything with that.
I can use numpy.typing as well, didn't know about that.
Maybe we should have a policy on type hints, so that we give them consistently in any new or modified code (if we want them at all, I think they're useful)?